plutus-pab/src/Plutus/PAB/Run.hs

```diff
@@ -67,19 +87,25 @@ runWith userContractHandler = do
     -- obtain token for signaling service readiness
     serviceAvailability <- newToken
 
+    pabConfig :: Either PABError Config <- case mc of
```

Comment:
This is a bit ugly; but I wasn't too sure of an immediately-obvious better idea that was worth the effort.

Things I thought about:

- Maybe somehow get AppOpts to use a `default` Config if the file isn't specified
- ???

Reply:
I think in this situation it would be nice if `AppOpts` had a type parameter like this

```haskell
data ConfigType = Spec | Concrete

type family LogConfig f where
  LogConfig Spec = Either FilePath LogConfig
  LogConfig Concrete = LogConfig

type family Config f where
  Config Spec = Either FilePath Plutus.PAB.Types.Config
  Config Concrete = Plutus.PAB.Types.Config

data AppOpts f = AppOpts { minLogLevel    :: Maybe Severity
                       , logConfig  :: LogConfig f
                       , config     :: Config f
                       , runEkgServer   :: Bool
                       , storageBackend :: StorageBackend
                       , cmd            :: Command
                       }

loadConfig :: AppOpts Spec -> Eff effs (AppOpts Concrete)
loadConfig AppOpts{logConfig, config} = -- load the values LogConfig and Plutus.PAB.Types.Config from the files if necessary
```

Then `parseOptions` would return a `AppOpts Spec` and all users of `AppOpts` would use `AppOpts Concrete`. And in the tests you could just construct the `AppOpts Conrete` directly

Reply:
You can leave that for a future PR though

Reply:
Here's a jira issue for that: https://jira.iohk.io/browse/SCP-2675